test/url-to-options.js

```diff
@@ -0,0 +1,56 @@
+import url from 'url';
+import test from 'ava';
+import urlToOptions from '../source/url-to-options';
+
+const githubURL = 'https://user:password@github.com:443/';
+
+test('converts node legacy URL to options', t => {
+	const parsedURL = url.parse(githubURL);
+	const options = urlToOptions(parsedURL);
+	const expected = {
+		hash: null,
+		hostname: 'github.com',
+		href: githubURL,
+		path: '/null',
+		pathname: '/',
+		port: 443,
+		protocol: 'https:',
+		search: null
+	};
+
+	t.deepEqual(options, expected);
+});
+
+test('converts URL to options', t => {
+	const parsedURL = new url.URL(githubURL);
+	const options = urlToOptions(parsedURL);
+	const expected = {
+		auth: 'user:password',
+		hash: '',
+		hostname: 'github.com',
+		href: 'https://user:password@github.com/',
+		path: '/',
+		pathname: '/',
+		protocol: 'https:',
+		search: ''
+	};
+
+	t.deepEqual(options, expected);
+});
+
+test('converts IPv6 URL to options', t => {
+	const IPv6URL = 'https://[2001:cdba::3257:9652]:443/';
+	const parsedURL = new url.URL(IPv6URL);
+	const options = urlToOptions(parsedURL);
+	const expected = {
+		hash: '',
+		hostname: '2001:cdba::3257:9652',
+		href: 'https://[2001:cdba::3257:9652]/',
+		path: '/',
+		pathname: '/',
+		protocol: 'https:',
+		search: ''
+	};
+
+	t.deepEqual(options, expected);
+});
```

Comment:
Needs another test with no port:
```js
const noPortURL = 'https://github.com/';
const parsedURL = new url.URL(noPortURL);
const options = urlToOptions(parsedURL);
const expected = {
    hash: '',
    hostname: 'github.com',
    href: 'https://github.com/',
    path: '/',
    pathname: '/',
    protocol: 'https:',
    search: ''
};
```

Reply:
Out of curiosity, why Reflect? Why not `hasOwnProperty`?

Reply:
Because `AVA`. (throws an error IIRC)

Reply:
> Out of curiosity, why Reflect? Why not hasOwnProperty?

`.hasOwnProperty` won't work with `Object.create(null)` created objects.

> Because AVA. (throws an error IIRC)

You mean `XO` ;)